contracts/StablecoinConverter.sol

```diff
@@ -108,4 +109,62 @@ contract StablecoinConverter is EpochTokenLocker {
     function tokenIdToAddressMap(uint16 id) public view returns (address) {
         return IdToAddressBiMap.getAddressAt(registeredTokens, id);
     }
-}
\ No newline at end of file
+
+    struct TradeData {
+        address owner;
+        uint volume;
+        uint16 orderId;
+    }
+
+    function submitSolution(
+        uint32 batchIndex,
+        address[] memory owner,  //tradeData is submitted as arrays
+        uint16[] memory orderId,
+        uint128[] memory volume,
+        uint128[] memory prices,  //list of prices for touched token only
+        uint16[] memory tokenIdForPrice  // price[i] is the price for the token with tokenID tokenIdForPrice[i]
+    ) public {
+        require(
+            batchIndex == getCurrentStateIndex() - 1,
+            "Solutions are no longer accepted for this batch"
+        );
```

Comment:
I think the validation is simply not needed. It does not provide any safety.

People can provide a false length array [a1, a2, ..., ak], even if we check for the size, by simply handing over [a1, a2,...ak, 0,...,0].
It would be security that could easily be circumvented. I hope that the limit price checks and token conservation checks will be sufficient to determine good solutions.

Reply:
I understand. Makes sense

Reply:
Incoherent lengths wouldn't be an issue if we passed a single array with packed bytes for each entry (20 bytes for address, 2 bytes for ownerId, 16 bytes for volume, etc), which could either be parsed into an array of TradeData struct or accessed from via helper methods (similar to [here](https://github.com/gnosis/dex-contracts/blob/on_chain_verifier/contracts/SnappAuctionChallenge.sol#L270-L273)). 

This would also save you variables.